### ISSUE TITLE
Create test fixtures that have nested targets delegation

### ISSUE THREAD

**Issue by tedbow** [php-tuf/php-tuf#142]
Currently the fixture set `TUFTestFixtureDelegated` has a delegation but we should have set that where the role delegated to delegates to another role. We could update the existing fixture or create a new one. 

IN https://github.com/php-tuf/php-tuf/pull/141 we add `testVerifiedDelegatedDownload()` we create this fixture we should use it in this test

**Comment by tedbow** [comment]
we created the fixture in #181(and fixed in #184)

The `TUFTestFixtureNestedDelegated` was for a correctly set up delegation chain.

To test some of the logic we are adding #141 we should create a couple other test fixtures after that is merged:
1. Add test fixture that has the problem we fixed in #184. Basically a level_1 delegation that has `paths: ['*.ext1']` and then level_1 delegates to level_2 which has `paths: ['*.ext2']` . There would be no files that match level_1 paths and also match level_2 paths so the updater shouldn't find any files in level_2
2. A level_1 delegation where `terminating: true` but then it still delegates to level_2. In this case level_2 should not be searched.
